plugins/amd-gpu/amd_gpu_test.py

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: LGPL-2.1+
+
+import ctypes
+import gi
+import os
+import sys
+import unittest
+from fwupd_test import FwupdTest, override_gi_search_path
+from fwupd_ioc import IOWR, IOW
+
+gi.require_version("UMockdev", "1.0")
+from gi.repository import UMockdev
+
+try:
+    override_gi_search_path()
+    gi.require_version("Fwupd", "2.0")
+    from gi.repository import Fwupd  # pylint: disable=wrong-import-position
+except ValueError:
+    # when called from unittest-inspector this might not pass, we'll fail later
+    # anyway in actual use
+    pass
+
+
+class drm_amdgpu_info_vbios(ctypes.Structure):
+    _fields_ = [
+        ("name", ctypes.c_char * 64),
+        ("vbios_png", ctypes.c_char * 64),
+        ("version", ctypes.c_uint32),
+        ("pad", ctypes.c_uint32),
+        ("vbios_ver_str", ctypes.c_char * 32),
+        ("date", ctypes.c_char * 32),
```

Comment:
it does seem a shame we can't generate these from the .rs files...

Reply:
split up